File: KF.Domain/Model/University.cs
using KF.Domain.Interface;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KF.Domain.Model
{
    public class University:Entity
    {
        public int UniversityId { get; set; }
        public string UniversityName { get; set; }
        public string UniversityAddress { get; set; }
        public string UniversityAbstract { get; set; }
        public string UniversityIcon { get; set; }
        public int CityId { get; set; }
        public int Undergraduate { get; set; }
        public int Postgraduate { get; set; }
        public int OverseasStudent { get; set; }
        public int Is211 { get; set; }
        public int Is985 { get; set; }
        public int IsDoubleClass { get; set; }

        public override void Initialize(Dictionary<string,string> filter)
        {
            if (filter.ContainsKey("universityId")) this.UniversityId = int.Parse(filter["universityId"]);
            if (filter.ContainsKey("universityName")) this.UniversityName = filter["universityName"];
            if (filter.ContainsKey("universityAbstract")) this.UniversityAbstract = filter["universityAbstract"];
            if (filter.ContainsKey("universityIcon")) this.UniversityIcon = filter["universityIcon"];
            if (filter.ContainsKey("universityAddress")) this.UniversityAddress = filter["universityAddress"];
            if (filter.ContainsKey("undergraduateTotalNumber")) this.Undergraduate = int.Parse(filter["undergraduateTotalNumber"]);
            if (filter.ContainsKey("postgraduateTotalNumber")) this.Postgraduate = int.Parse(filter["postgraduateTotalNumber"]);
            if (filter.ContainsKey("overseasStudentTotalNumber")) this.OverseasStudent = int.Parse(filter["overseasStudentTotalNumber"]);
            if (filter.ContainsKey("cityId")) this.CityId = int.Parse(filter["cityId"]);
            if (filter.ContainsKey("Is211")) this.Is211 = int.Parse(filter["Is211"]);
            if (filter.ContainsKey("Is985")) this.Is985 = int.Parse(filter["Is985"]);
            if (filter.ContainsKey("IsDoubleClass")) this.IsDoubleClass = int.Parse(filter["IsDoubleClass"]);
        }
    }
}

File: KF.UniversityWeb/Scripts/Framework/university.js
function university() {
    this.universityId = 0;
    this.universityName = "";
    this.universityAddress = "";
    this.universityAbstract = "";
    this.universityIcon = "";
    this.cityId = 0;
    this.undergraduateTotalNumber = 0;
    this.postgraduateTotalNumber = 0;
    this.overseasStudentTotalNumber = 0;
    this.Is211 = 0;
    this.Is985 = 0;
    this.IsDoubleClass = 0;

    this.loadObservers = [];
}

university.prototype.load = function () {
    return $.ajax({
        url: "/universityService",
        data: {
            universityId: this.universityId,
            option: "getone"
        },
        success: $.proxy(function (ret) {
            this.universityId = ret.UniversityId;
            this.universityName = ret.UniversityName;
            this.universityAddress = ret.UniversityAddress;
            this.cityId = ret.CityId;
            var abstract = $.decodeUTF8(ret.UniversityAbstract);
            this.universityAbstract = abstract;
            this.universityIcon = ret.UniversityIcon;
            this.undergraduateTotalNumber = ret.Undergraduate;
            this.postgraduateTotalNumber = ret.Postgraduate;
            this.overseasStudentTotalNumber = ret.OverseasStudent;
            this.Is211 = ret.Is211;
            this.Is985 = ret.Is985;
            this.IsDoubleClass = ret.IsDoubleClass;


            var count = this.loadObservers.length;
            for (var i = 0; i < count; i++) {
                this.loadObservers[i](this);
            }
        }, this)
    });
}

university.prototype.addLoadObserver = function (observer) {
    this.loadObservers.push(observer);
}


function universitys() {
    kf.util.entitiesPage.call(this);
    this.filter.provinceId = 0;
    this.filter.cityId = 0;
    this.loadObservers = [];
}

$.extend(universitys.prototype, kf.util.entitiesPage.prototype);

universitys.prototype.load = function () {
    $.ajax({
        url: "/universityService",
        data: {
            pageIndex: this.filter.pageIndex,
            pageSize: this.filter.pageSize,
            cityId: this.filter.cityId,
            provinceId: this.filter.provinceId,
            universityName: this.filter.universityName || "",
            option: "getlist"
        },
        success: $.proxy(function (ret) {
            this.clear();
            this.totalCount = ret.TotalCount;
            var count = ret.Values.length;
            for (var i = 0; i < count; i++) {
                var _university = new university();
                _university.universityId = ret.Values[i].UniversityId;
                _university.universityName = ret.Values[i].UniversityName;
                _university.universityIcon = ret.Values[i].UniversityIcon;
                _university.universityAddress = ret.Values[i].UniversityAddress;
                _university.cityId = ret.Values[i].CityId;
                this.add(_university.universityId, _university);
            }

            count = this.loadObservers.length;
            for (var i = 0; i < count; i++) {
                this.loadObservers[i](this);
            }
        }, this)
    });
}

universitys.prototype.addLoadObserver = function (observer) {
    this.loadObservers.push(observer);
}